lib/resolve.js

```diff
@@ -29,6 +29,17 @@ module.exports = unyield(function* resolve(slug, options) {
   if (!parsed.ref) parsed.ref = '*';
   var ref = parsed.ref;
 
+  if (isSHA(ref)) {
+    debug('%s: resolving via commit SHA', slug, ref);
+    try {
```

Comment:
Oh, I think I understand now. We're checking if a "sha-like" string actually references a commit. If a commit w/ that SHA doesn't exist, it throws. (but we want it to check tags/branches in case we wrongly guessed it was supposed to be a commit.

Never mind, this lgtm!